### PR TITLE
updated mlag heartbeat VRF logic

### PR DESCRIPTION
**Pull request by mthiel117**
## Change Summary
Update logic for MLAG configuration section for peer heartbeat to allow default and non-default VRFs.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## How to test
run playbook generate EOS configs and verified that default and non-default VRFs create the correct configurations.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mlag-configuration.j2 [hunk 1]
```diff
@@ -6,9 +6,12 @@ mlag configuration
    local-interface {{ mlag_configuration.local_interface }}
    peer-address {{ mlag_configuration.peer_address }}
 {%   if mlag_configuration.peer_address_heartbeat.peer_ip is defined and mlag_configuration.peer_address_heartbeat.peer_ip is not none %}
-   peer-address heartbeat {{ mlag_configuration.peer_address_heartbeat.peer_ip }}{% if mlag_configuration.peer_address_heartbeat.vrf is defined and mlag_configuration.peer_address_heartbeat.vrf is not none and mlag_configuration.peer_address_heartbeat.vrf is ne 'default' %} vrf {{ mlag_configuration.peer_address_heartbeat.vrf }}
-{%       endif -%}
-{%   endif %}
+{%   if mlag_configuration.peer_address_heartbeat.vrf is defined and mlag_configuration.peer_address_heartbeat.vrf is not none and mlag_configuration.peer_address_heartbeat.vrf is ne 'default' %}
+   peer-address heartbeat {{ mlag_configuration.peer_address_heartbeat.peer_ip }} vrf {{ mlag_configuration.peer_address_heartbeat.vrf }}
+{%    else %} {# using the default VRF #}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mlag-configuration.j2]: Please move comment above the `{% else %}` statement, as opposed to in-line as per below. The space between the `{% else %}` statement and comments results in adding one extra whitespace on the command when rendered.
 
```
{# using the default VRF #}
{%    else %}
```

**Review by carlbuchmann — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mlag-configuration.j2 (hunk 1 above)
Comment (mthiel117) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mlag-configuration.j2]: Change completed

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
- [x] update indentation of statements for code readability
- [x] remove strip whitespace in two statements `{%      endif -%}`, which causes undesired results in the generated configuration

example -> peer-link Port-Channel5 not indented correctly:
```
mlag configuration
   domain-id DC1_BL1
   local-interface Vlan4094
   peer-address 10.255.252.11
   peer-address heartbeat 192.168.200.111 vrf MGMT
peer-link Port-Channel5
   dual-primary detection delay 5 action errdisable all-interfaces
   reload-delay mlag 300
   reload-delay non-mlag 330
```

Resolved template:

```
{% if mlag_configuration is defined and mlag_configuration is not none %}
!
mlag configuration
   domain-id {{ mlag_configuration.domain_id }}
   local-interface {{ mlag_configuration.local_interface }}
   peer-address {{ mlag_configuration.peer_address }}
{%    if mlag_configuration.peer_address_heartbeat.peer_ip is defined and mlag_configuration.peer_address_heartbeat.peer_ip is not none %}
{%        if mlag_configuration.peer_address_heartbeat.vrf is defined and mlag_configuration.peer_address_heartbeat.vrf is not none and mlag_configuration.peer_address_heartbeat.vrf is ne 'default' %}
   peer-address heartbeat {{ mlag_configuration.peer_address_heartbeat.peer_ip }} vrf {{ mlag_configuration.peer_address_heartbeat.vrf }}
{# using the default VRF #}
{%        else %}
   peer-address heartbeat {{ mlag_configuration.peer_address_heartbeat.peer_ip }}
{%        endif %}
{%    endif %}
   peer-link {{ mlag_configuration.peer_link }}
{%    if mlag_configuration.dual_primary_detection_delay is defined and mlag_configuration.dual_primary_detection_delay is not none %}
   dual-primary detection delay {{ mlag_configuration.dual_primary_detection_delay }} action errdisable all-interfaces
{%    endif %}
   reload-delay mlag {{ mlag_configuration.reload_delay_mlag }}
   reload-delay non-mlag {{ mlag_configuration.reload_delay_non_mlag }}
{% endif %}
```

**Comment by mthiel117** [comment]
Updated template to fix whitespace issue that impacted indentation under mlag configuration section when using default or non-default VRF.

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**